packages/calcite-components/src/components/list/list.tsx

```diff
@@ -813,11 +811,12 @@ export class List
     this.updateFilteredData();
   }
 
-  private getItemData(): ItemData {
+  private getItemData(): ItemData[] {
     return this.listItems.map((item) => ({
       label: item.label,
       description: item.description,
       metadata: item.metadata,
+      heading: item.parentElement?.closest(listItemGroupSelector)?.heading || "",
```

Comment:
Can we store the parentGroup elements on connectedCallback? That way its only ever queried in the DOM one time. In this scenario its queried every time `getItemData()` is called.

The other part is that this is only getting the first parent group element. Shouldn't it get all ancestor group elements and map to their headings?

Reply:
>The other part is that this is only getting the first parent group element. Shouldn't it get all ancestor group elements and map to their headings?

Good catch. Each item should store the heading of all ancestors group elements.